utils/token.ts

```diff
@@ -0,0 +1,34 @@
+import { AccessToken, Prisma, PrismaClient } from "@prisma/client";
+
+type TokenArgs = Prisma.AccessTokenUncheckedCreateInput;
+
+const prisma = new PrismaClient();
+
+/**
+ * Creates a new token if there is not token with the "associatedEmail" field.
+ * More on prisma's upsert method here: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upserthttps://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert
+ *
+ * @param token {TokenArgs} - Token that has to be created/updated.
+ * */
+const upsertToken = async (data: TokenArgs): Promise<AccessToken | Error> => {
+  try {
+    const upsertData = await prisma.accessToken.upsert({
+      where: {
+        AccessToken_constraint: {
+          calendarId: data.calendarId,
+          calendarType: data.calendarType,
+          userId: data.userId,
+        },
+      },
+      create: data,
+      update: data,
+    });
+
+    return upsertData;
+  } catch (error) {
+    logger.error(error);
+    throw new Error(JSON.stringify(error));
```

Comment:
What if we simply `throw error` ?

Reply:
Gives me this error: `Argument of type 'unknown' is not assignable to parameter of type 'string | undefined'.ts(2345)`